### PR TITLE
WIP: Initial job docs

### PR DESCRIPTION
**Pull request by stgricci**
Adding initial spec documentation for JobTargetRef ScaledObject

### CODE REVIEW

**Comment by stgricci** [comment]
/cc @tomkerkhove @ahmelsayed 

Ref: #430

File: spec/ScaledObject.md [hunk 1]
```diff
@@ -36,6 +45,25 @@ The name of the deployment this scaledObject is for. This is the deployment keda
 
 **Assumptions:** `deploymentName` is in the same namespace as the scaledObject
 
+---
```
Comment (tomkerkhove) [inline spec/ScaledObject.md]: Don't think we need this, right?
```suggestion
```

**Comment by tomkerkhove** [comment]
THanks @stgricci - LGTM, but this does not align with #430.
I guess #430 is wrong then or?

**Comment by ahmelsayed** [comment]
looking at the code, I think #430 is correct and this PR is missing a `spec:`
`jobTargetRef.template` is a `JobSpec` [`v1.PodTemplateSpec`](https://godoc.org/k8s.io/api/core/v1#PodTemplateSpec) which takes a 

```js
{
  "metadata": {
    // v1.Metadata
  },
  "spec": {
    // job spec
  }
}
```

**Comment by tomkerkhove** [comment]
Sounds good to me!

**Comment by stgricci** [comment]
Yep, @tomconte was correct. I'll update these docs and then we can merge them in

**Comment by tomkerkhove** [comment]
I'll be moving this to our doc site and is tracked here: https://github.com/kedacore/keda-docs/issues/7

**Comment by jeffhollan** [comment]
Moved docs over here https://github.com/kedacore/keda-docs/pull/22

**Pull request closed without merging**